scripts/test_metric_reporter/config.py

```diff
@@ -0,0 +1,117 @@
+
+import logging
+import os
+import re
+
+from configparser import ConfigParser, NoSectionError, NoOptionError
+from pydantic import BaseModel, ValidationError, Field
+
+from scripts.common.config import BaseConfig, InvalidConfigError, DIRECTORY_PATTERN
+
+
+class TestMetricArgs(BaseModel):
+    """Model for test metric arguments."""
+
+    test_metadata_directory_path: str
+    csv_report_file_path: str
+
+
+class TestMetricReporterConfig(BaseModel):
+    """Model for test metric reporter configuration."""
+
+    reports_dir: str = Field(..., pattern=DIRECTORY_PATTERN)
+
+
+class Config(BaseConfig):
+    """Configuration handler for the Test Metric Reporter."""
+
+    logger = logging.getLogger(__name__)
+
+    def __init__(self, config_file: str = "config.ini") -> None:
+        """Initialize the Config.
+
+        Args:
+            config_file (str): Path to the configuration file.
+
+        Raises:
+            InvalidConfigError: If the configuration file contains missing or invalid values,
+                                or if an error occurs while building test metric reporter arguments.
+        """
+        super().__init__(config_file)
+        self.test_metric_reporter_config: TestMetricReporterConfig = (
+            self._parse_test_metric_reporter_config(self.config_parser)
+        )
+        self.test_metric_reporter_args: list[TestMetricArgs] = (
+            self._build_test_metric_reporter_args()
+        )
+        self.logger.info("Successfully loaded configuration")
+
+    def _parse_test_metric_reporter_config(
+        self, config_parser: ConfigParser
+    ) -> TestMetricReporterConfig:
+        try:
+            reports_dir: str = config_parser.get("test_metric_reporter", "reports_dir")
+            return TestMetricReporterConfig(reports_dir=reports_dir)
+        except (NoSectionError, NoOptionError, ValidationError) as error:
+            error_mapping: dict[type, str] = {
+                NoSectionError: "The 'test_metric_reporter' section is missing",
+                NoOptionError: "Missing config option in 'test_metric_reporter' section",
+                ValidationError: "Unexpected value or schema in 'test_metric_reporter' section",
+            }
+            error_msg = error_mapping[type(error)]
+            self.logger.error(error_msg, exc_info=error)
+            raise InvalidConfigError(error_msg, error)
+
+    @staticmethod
+    def _normalize_name(name: str, delimiter: str = "") -> str:
+        normalized = re.sub(r"[^a-zA-Z0-9_]+", delimiter, name).lower()
+        return normalized.strip("_")
+
+    def _build_test_metric_reporter_args(self) -> list[TestMetricArgs]:
+        # Here we assume that the structure of the test_result_dir directory is as follows:
+        # test_result_dir/
+        #     ├── repository/
+        #         ├── workflow/
+        #             ├── test_suite/
+        #                 ├── test_metadata_dir/
+        try:
```

Comment:
I am a bit confused on what this does, could you explain it a bit?

Reply:
Are we expecting this to change? Seems like we could hard code it?

Reply:
Yes we are expecting this to change and soon! Maybe wait till it gets bigger and we can re-visit organizing the data.